src/globalvars.cpp

```diff
@@ -96,19 +121,30 @@ void update_global_vars_rodata(
 
   // Update the values for the global vars (using the above offsets)
   for (const auto &[global_var, offset] : vars_and_offsets) {
-    auto *var = reinterpret_cast<int64_t *>(global_vars_buf + offset);
-
-    auto it = known_global_var_values.find(global_var);
-    if (it != known_global_var_values.end()) {
-      *var = it->second;
+    auto it = global_var_vals.find(global_var);
+    if (it != global_var_vals.end()) {
+      const auto &config = added_global_vars.at(global_var);
+      if (config.type == Type::integer) {
```

Comment:
Maybe use `switch` instead of this `if`-`else if`-`else`? We'd get some compiler checks for free.

Reply:
I considered that but we're only using a small subset of the `Type` enum from sized type and so we'd have A LOT of boilerplate to get some checks about non-exhaustive type checks.

Reply:
My idea was that we'd handle the rest via `default` but in that case, it probably doesn't matter if we use `if` chain or `switch`.